mesonbuild/dependencies/base.py

```diff
@@ -655,6 +655,12 @@ def setup_env(env, environment, for_machine, extra_path=None):
         sysroot = environment.properties[for_machine].get_sys_root()
         if sysroot:
             env['PKG_CONFIG_SYSROOT_DIR'] = sysroot
+        # Add the pkgconfig directory from the installation prefix to PKG_CONFIG_PATH:
+        if environment.coredata.builtins_per_machine[for_machine]['default_dependency_sources'].value in ('install', 'both'):
```

Comment:
Why accept it if it's not handled the way you'd expect it to be? Actually setting this to "install" would result in behavior which is surprising now, and not backwards-compatible if it gets implemented later.